langlib/langlib-test/src/test/resources/test-src/tablelib_test.bal

```diff
@@ -683,3 +683,37 @@ function testPutValidDataToKeylessTbl() returns boolean {
     testPassed = testPassed && tableToList[2] == intern1;
     return testPassed;
 }
+
+function testReadOnlyTableFilter() {
+    PersonalTable & readonly personTable = table [
+      { name: "Harry", age: 14 },
+      { name: "Hermione", age: 28 },
+      { name: "Ron", age: 11 },
+      { name: "Draco", age: 23 }
+    ];
+    table<Person> children = personTable.filter(function (Person person) returns boolean {
+                                                      return person.age < 18;
+                                                  });
+    assertEquals(children.length(), 2);
+    children.forEach(function(Person person) {
+        assertTrue(person.age < 18);
+    });
+}
+
+const ASSERTION_ERROR_REASON = "AssertionError";
+
+function assertTrue(boolean actual) {
+    assertEquals(true, actual);
+}
+
+function assertEquals(anydata expected, anydata actual) {
```

Comment:
I think we can use function in `ballerina/test` package instead of defining our own functions in all places

Reply:
We usually try to avoid importing non-langlib modules in language tests. Once the compiler test framework (most of the  changes done in https://github.com/ballerina-platform/ballerina-lang/pull/24557) is fully integrated, the plan is to migrate all of these lang tests to that.

IMO, we can go ahead with this approach for the time-being.